### ISSUE TITLE
Ensure router properly replaces components via /?next=

### ISSUE THREAD

**Issue by taoeffect** [okTurtles/group-income#127]
### Problem

Vue components seem to accrue.

From a fresh `grunt dev` run, do:

1. Visit http://localhost:8000/simple/
2. Click "Start a group", you'll be redirected to http://localhost:8000/simple/signup?next=%2Fnew-group
3. Click "Pay Group"
4. Click "Start a group" again

In the Chrome Vue devtools, you'll now see two `<CreateGroupView>`s:

![screen shot 2016-11-27 at 8 21 49 pm](https://cloud.githubusercontent.com/assets/138706/20656217/3568a5a4-b4df-11e6-8884-c1a250e8fbf3.jpg)

If you repeat steps 3 and 4 the `CreateGroupView` components will just keep stacking.

### Solution

Fix it so that components are properly destroyed.

**Comment by taoeffect** [comment]
This is also a part of / continuation of #93.

**Comment by taoeffect** [comment]
Oops, this has nothing to do with VueAssistant, but rather with my usage of vue-router. I don't know if this problem/code is affected by the upgrade to vue-router 2.

Taken from SignUp.vue:

```js
    submit: function () {
      this.response = ''
      request.post(`${process.env.API_URL}/user/`)
      .send(serialize(this.form, {hash: true}))
      .end((err, res) => {
        this.error = !!err || !res.ok
        console.log('this.error', this.error)
        this.response = this.error ? res.body.message : (res.text === '' ? 'success' : res.text)
        if (!this.error && this.$route.query.next) {
          this.login()
          this.$route.router.go({path: this.$route.query.next})
        }
      })
    },
```

**Comment by vijayee** [comment]
This should be corrected as of #129
